sdk/examples/devmode_rust/src/engine.rs

```diff
@@ -72,15 +72,24 @@ impl DevmodeService {
 
     fn finalize_block(&mut self) {
         debug!("Finalizing block");
-        let mut query_result = self.service.finalize_block(Vec::from(&b"Devmode"[..]));
+        let mut summary = self.service.summarize_block();
+        while let Err(Error::BlockNotReady) = summary {
+            warn!("Block not ready");
```

Comment:
Maybe these warning should specify "Block not ready for summary" and "Block not ready for finalizing".